File: db/view_event_person.php
<?php

require_once("view.php");

/**
 * Class for accessing, manipulating, creating and deleting records in event_person.
*/

abstract class View_Event_Person extends View
{

  public function __construct($select = array())
  {
    $this->domain = "event";
    $this->field['event_person_id']['type'] = 'INTEGER';
    $this->field['event_person_id']['table'] = 'event_person';
    $this->field['event_id']['type'] = 'INTEGER';
    $this->field['event_id']['table'] = 'event_person';
    $this->field['person_id']['type'] = 'INTEGER';
    $this->field['person_id']['table'] = 'event_person';
    $this->field['event_role_id']['type'] = 'INTEGER';
    $this->field['event_role_id']['table'] = 'event_person';
    $this->field['event_role_state_id']['type'] = 'INTEGER';
    $this->field['event_role_state_id']['table'] = 'event_person';
    $this->field['remark']['type'] = 'TEXT';
    $this->field['remark']['table'] = 'event_person';
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['rank']['table'] = 'event_person';
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/views/view_event_person.rb
module Momomoto
  class View_event_person < Base
    def initialize
      super
      @domain = 'event'
      @fields = {
        :event_person_id => Datatype::Integer.new( {} ),
        :event_id => Datatype::Integer.new( {} ),
        :person_id => Datatype::Integer.new( {} ),
        :event_role_id => Datatype::Integer.new( {} ),
        :event_role_state_id => Datatype::Integer.new( {} ),
        :remark => Datatype::Text.new( {} ),
        :rank => Datatype::Integer.new( {} ),
        :conference_id => Datatype::Integer.new( {} ),
        :title => Datatype::Varchar.new( {:length=>128} ),
        :subtitle => Datatype::Varchar.new( {:length=>128} ),
        :event_state_id => Datatype::Integer.new( {} ),
        :name => Datatype::Varchar.new( {} ),
        :language_id => Datatype::Integer.new( {} ),
        :language_tag => Datatype::Varchar.new( {:length=>32} ),
        :event_state_tag => Datatype::Varchar.new( {:length=>32} ),
        :event_state => Datatype::Varchar.new( {} ),
        :event_role_tag => Datatype::Varchar.new( {:length=>32} ),
        :event_role => Datatype::Varchar.new( {} ),
        :event_role_state_tag => Datatype::Varchar.new( {:length=>32} ),
        :event_role_state => Datatype::Varchar.new( {} )
      }
    end
  end
end
